pandas/tests/indexes/test_old_base.py

```diff
@@ -223,11 +223,10 @@ def test_logical_compat(self, simple_index):
             assert idx.any() == idx.to_series().any()
         else:
             msg = "cannot perform (any|all)"
```

Comment:
I think you can delete L225-L228? i.e. the only thing needed here to pass tests is `msg = "does not support reduction '(any|all)'"`?

Reply:
thanks, I deleted the branch with `msg = "datetime64 type does not support operation: '(any|all)'"`, ci- green.

Reply:
Oh OK I see the issue now. So you are branching because DatetimeIndex says `does not support operation: '(any|all)'` but the other cases say `'does not support reduction '(any|all)'`

I don't want to go down a rabbit hole but having a message that differs between **operation:** and **reduction** doesn't seem that important. How hard is it to align them so you can just assign `msg = "does not support operation: '(any|all)'` and have the tests pass?